### PR TITLE
Fix triple default version

### PR DESCRIPTION
**Pull request by AlbumenJ**
## What is the purpose of the change

Fix triple protocol to treat `null` and `1.0.0` the same.
If you want triple work as before, set `dubbo.rpc.tri.ignore-1.0.0-version` and `dubbo.rpc.tri.resolve-fallback-to-default` are `true` in jvm properties.

## Checklist
- [x] Make sure there is a [GitHub_issue](https://github.com/apache/dubbo/issues) field for the change (usually before you start working on it). Trivial changes like typos do not require a GitHub issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue.
- [ ] Each commit in the pull request should have a meaningful subject line and body.
- [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
- [ ] Check if is necessary to patch to Dubbo 3 if you are work on Dubbo 2.7
- [ ] Write necessary unit-test to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add sample in [dubbo samples](https://github.com/apache/dubbo-samples) project.
- [ ] Add some description to [dubbo-website](https://github.com/apache/dubbo-website) project if you are requesting to add a feature.
- [ ] GitHub Actions works fine on your own branch.
- [ ] If this contribution is large, please follow the [Software Donation Guide](https://github.com/apache/dubbo/wiki/Software-donation-guide).

### CODE REVIEW

**Comment by guohao** [comment]
PLZ do NOT modify these behaviors, It is a protocol spec that  '1.0.0'  or 'null' version MUST not passed to transport layer to coupe with mixing-framework envs like grpc/sofa .

File: dubbo-rpc/dubbo-rpc-triple/src/main/java/org/apache/dubbo/rpc/protocol/tri/RequestMetadata.java [hunk 1]
```diff
@@ -58,9 +58,7 @@ public DefaultHttp2Headers toHeaders() {
             .set(TripleHeaderEnum.CONTENT_TYPE_KEY.getHeader(), TripleConstant.CONTENT_PROTO)
             .set(HttpHeaderNames.TE, HttpHeaderValues.TRAILERS);
         setIfNotNull(header, TripleHeaderEnum.TIMEOUT.getHeader(), timeout);
-        if (!"1.0.0".equals(version)) {
-            setIfNotNull(header, TripleHeaderEnum.SERVICE_VERSION.getHeader(), version);
-        }
```
Comment (chickenlj) [inline dubbo-rpc/dubbo-rpc-triple/src/main/java/org/apache/dubbo/rpc/protocol/tri/RequestMetadata.java:63]: What's the purpose of this check? Why need to set NULL for `1.0.0`?

**Comment by chickenlj** [comment]
> PLZ do NOT modify these behaviors, It is a protocol spec that '1.0.0' or 'null' version MUST NOT pass to transport layer to coupe with mixing-framework envs like grpc/sofa .

Any further background on why  '1.0.0'  should be specially treated in the SPEC, it's a bit weird why it has to be 1.0.0?

I think gRPC would simply ignore the `version` part since it has no concept of version.

File: dubbo-rpc/dubbo-rpc-triple/src/main/java/org/apache/dubbo/rpc/protocol/tri/stream/TripleServerStream.java [hunk 2]
```diff
@@ -287,14 +287,7 @@ private Invoker<?> getInvoker(Http2Headers headers, String serviceName) {
             headers.contains(TripleHeaderEnum.SERVICE_GROUP.getHeader()) ? headers.get(
                 TripleHeaderEnum.SERVICE_GROUP.getHeader()).toString() : null;
         final String key = URL.buildKey(serviceName, group, version);
-        Invoker<?> invoker = pathResolver.resolve(key);
-        if (invoker == null) {
```
[the remark this answers is not in the document]
Comment (guohao) [reply dubbo-rpc/dubbo-rpc-triple/src/main/java/org/apache/dubbo/rpc/protocol/tri/stream/TripleServerStream.java:291]: `VERSION` and `GROUP` has  a weak semantics in Triple protocol for interoperation.

**Comment by guohao** [comment]
It is probably better to treat 'triple protocol' in `AS IS`  semantics.  It certainly has different behaviors with 'dubbo' protocol and may not apply to the 'widely adopted and recognized design and behaviors' .

File: dubbo-rpc/dubbo-rpc-triple/src/main/java/org/apache/dubbo/rpc/protocol/tri/RequestMetadata.java (hunk 1 above)
Comment (CrazyHZM) [reply dubbo-rpc/dubbo-rpc-triple/src/main/java/org/apache/dubbo/rpc/protocol/tri/RequestMetadata.java:63]: Why do I get the feeling that this 1.0.0 is a bit confusing with the Dubbo protocol version?
If it's just the service version number, it shouldn't make a difference to 1.0.0.

**Review by chickenlj — APPROVED** [review]
LGTM.

**Pull request merged**